### PR TITLE
Update clp-ffi-py dependency; Drop support for Python3.6.

### PR DESCRIPTION
**Pull request by LinZhihao-723**
# References

# Description

`clp-ffi-py` 0.0.11 release adds support for Windows build. This PR updates `clp-ffi-py` dependency for future Windows support. Since the latest `clp-ffi-py` has dropped support for Python3.6, this PR should no longer support Python3.6 either.
To fully support Windows build, we will still need a few changes to fix platform-dependent APIs in the coming PRs. Thus, we will not automate Windows build/test in our GitHub workflow until we have all the things fixed.

# Validation performed

Passed existing GitHub workflows.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,9 @@ capabilities.
 
 For a detailed understanding of the CLP IR format, refer to [README-protocol.md](README-protocol.md)
 
+> [!IMPORTANT]
+> This project is no longer built for Python3.6.
+
```
Comment (davidlion) [inline README.md]: I feel like we don't need this considering 3.6 is pretty outdated. Even 3.7 has been end-of-life for some time.

File: README.md [hunk 2]
```diff
@@ -208,7 +211,7 @@ logging.getLogger(__name__).addHandler(clp_handler)
 
 ## Compatibility
 
-Tested on Python 3.6, 3.8, and 3.11 (should also work on newer versions).
+Tested on Python 3.8 and 3.11 (should also work on newer versions).
```
Comment (davidlion) [inline README.md]: I think we can edit/reword this section to explicitly say our minimum supported version is 3.7. We should probably test on 3.7 considering that is the minimum required version in pyproject.toml.

**Review by davidlion — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (LinZhihao-723) [reply README.md]: We don't need to explicitly state the minimum supported version. It will be shown in the PyPI project home page :D

**Review by davidlion — APPROVED** [review]

**Review by davidlion — APPROVED** [review]
title fine for commit msg

**Pull request merged**